src/test/clojure/clara/test_durability.clj

```diff
@@ -379,3 +380,27 @@
 
     (testing "restoring with memory"
       (test-assemble orig true))))
+
+;; Issue 422 (https://github.com/cerner/clara-rules/issues/422)
+;; A test to demonstrate the difference in error messages provided when compilation context is omitted
+(deftest test-compilation-ctx
+  (def some-var 123)
+  (let [rule (dsl/parse-rule [[Long (== this some-var)]]
+                             (println "here"))
+        without-compile-ctx (com/mk-session [[rule]])
+        with-compile-ctx (com/mk-session [[rule] :omit-compile-ctx false])]
+    (ns-unmap 'clara.test-durability 'some-var)
+    (try
+      (rb-serde without-compile-ctx nil)
+      (is false "Error not thrown when deserializing the rulebase without ctx")
+      (catch Exception e
+        (is (= (.getMessage e)
+               "Failed compiling.\n{:expr (clojure.core/fn [?__fact__ ?__env__] (clojure.core/let [this ?__fact__ ?__bindings__ (clojure.core/atom {})] (if (== this clara.test-durability/some-var) (clojure.core/deref ?__bindings__) nil)))}\n"))))
```

Comment:
It's be nice to regex this perhaps instead of dump the literal output.
The thing is that this could easily change with future compiler work and having to fix things like this can be annoying.

Reply:
+1, this should be made less brittle somehow IMO.  I tend to prefer looking at the ex-info data map rather than the message where applicable and requiring that some Throwable in the stack match the predicate, not that the top one do so, in case the exception is later caught and re-thrown as the cause of something else.